code/modules/clothing/glasses/_glasses.dm

```diff
@@ -307,7 +307,15 @@
 	name = "beer goggles"
 	icon_state = "sunhudbeer"
 	desc = "A pair of sunglasses outfitted with apparatus to scan reagents, as well as providing an innate understanding of liquid viscosity while in motion."
-	clothing_traits = list(TRAIT_BOOZE_SLIDER, TRAIT_REAGENT_SCANNER)
+	clothing_traits = list(TRAIT_DRUNK_VISION)
+
+/obj/item/clothing/glasses/sunglasses/reagent/equipped(mob/living/carbon/human/user, slot)
+	. = ..()
+	user.update_eye_blur()
```

Comment:
You need to check if it was equipped in the right slot in `equipped`, or else it'll apply when in hand

Reply:
I saw that with other glasses but `update_eye_blur` doesn't do anything if you don't have the DRUNK TRAIT.  Regardless I will make it consistent with the other glasses.  Applied with 4101513c2f08147570369c89352774779cc7e556